### PR TITLE
litemdview: init at 0.0.32

### PR DESCRIPTION
**Pull request by WhiteBlackGoose**
Minimalist suckless markdown viewer. Serves
the same purpose and people as nsxiv, zathura.

###### Description of changes

This is a new package: [litemdview](https://notabug.org/g0tsu/litemdview). Amazing suckless markdown viewer, similar to zathura and nsxiv in its purpose.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I haven't found any new packages in `rl-2305.section.md` file, so not sure if I should add this one to any release notes.

Tested the correctness as follows:
```
nix-shell -p litemdview -I nixpkgs=~/prj/nixpkgs/nixpkgs
```

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (NobbZ) [inline pkgs/top-level/all-packages.nix:32322]:
```suggestion
```

File: maintainers/maintainer-list.nix
Comment (NobbZ) [inline maintainers/maintainer-list.nix:16189]: Please make this a seperate commit **before** the one that adds the package

**Review by NobbZ — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:15]: Please use a proper version or `unstable-$commitDate` (I think this is current consensus)

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:25]: Some of your current `buildInputs` should be `nativeBuildInputs`, on a first glance at least `autoconf`, `automake`, `pkg-config`.

**Review by NobbZ — CHANGES_REQUESTED** [review]

**Review by NobbZ — CHANGES_REQUESTED** [review]
Please make sure to name your commits properly.

1. `maintainers: Add $you as a maintainer`
2. `litemdview: init at $version` (preferably accoring to current version preference for non-released software)

**Comment by WhiteBlackGoose** [comment]
Thanks. Apparently there *is* a version, I just never noticed it. Took into account all your comments, hopefully

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:35]: I don't know `libtool` and `cairo` well enough, though are you sure that these are `nativeBuildInputs`?

Which roughly means, they provide binaries that are required to run during compilation, rather than libraries that need to be linked and available during runtime?

This distinction is mostly useful for cross compiling, still should be done correctly.

If you can explain why they should be `nativeBuildInputs` I'm fine with that…

PS: Crosscompiling for `aarch64` worked though, and I was also able to run via `qemu`, therefore I am really not sure if this is correct or not. This probably has to be answered by someone who understands the crosscompiling better than me.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:40]: Not sure what the `./bootstrap` does, though `./configure` is traditionally ran during `configurePhase`, and might already happen by the default phase as provided by `stdenv.mkDerivation`.

Please check the necessity of these being need to be run explicitely and move them to the correct phase.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:41]: You should be able to omit the `buildPhase`, if `make` is the sole command to be run.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:44]: You can omit the `installPhase` if `make install` is the sole command to be run.

**Review by NobbZ — CHANGES_REQUESTED** [review]

**Comment by NobbZ** [comment]
Oh, and just realising, if it is inspired by suckless, is it also "configure through patches"?

In this case an overridable `patches ? []` would make sense.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (tejing1) [inline pkgs/applications/graphics/litemdview/default.nix:39]: `./bootstrap` should go in `preConfigure`, then you can let stdenv's normal phases take it from there.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (WhiteBlackGoose) [reply pkgs/applications/graphics/litemdview/default.nix:35]: :warning:  Not sure about this one

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 209770` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>litemdview</li>
  </ul>
</details>

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [reply pkgs/applications/graphics/litemdview/default.nix:35]: As `cairo` felt very weird being here, I tried removing it completely from the inputs. The build suceeds even without, are you sure that `cairo` is a required dependency?

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:40]:
```suggestion
```

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:38]:
```suggestion
  preConfigure = "./bootstrap";
```

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [reply pkgs/applications/graphics/litemdview/default.nix:35]:
```suggestion
```

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NobbZ) [inline pkgs/applications/graphics/litemdview/default.nix:9]:
```suggestion
```

**Review by NobbZ — CHANGES_REQUESTED** [review]
Seems as if `cairo` really isn't necessary.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (WhiteBlackGoose) [reply pkgs/applications/graphics/litemdview/default.nix:35]: Strange, it didn't build for me without cairo
Maybe it wasn't cairo that was the problem, will try without it

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 209770` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>litemdview</li>
  </ul>
</details>

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 209770` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>litemdview</li>
  </ul>
</details>

**Review by NobbZ — APPROVED** [review]

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NickCao) [inline pkgs/applications/graphics/litemdview/default.nix:36]: What about trying autoreconfHook?

Comment (tejing1) [reply pkgs/applications/graphics/litemdview/default.nix:36]: Indeed, `autoreconfHook` would be more idiomatic. That's all the bootstrap script seems to do, so we can drop it and remove its dependencies from `nativeBuildInputs`.

**Comment by tejing1** [comment]
```diff
diff --git a/pkgs/applications/graphics/litemdview/default.nix b/pkgs/applications/graphics/litemdview/default.nix
index 12939c7b183..36864b4a1c3 100644
--- a/pkgs/applications/graphics/litemdview/default.nix
+++ b/pkgs/applications/graphics/litemdview/default.nix
@@ -3,9 +3,7 @@
 , fetchFromGitea
 , fetchpatch
 , gtkmm3
-, autoconf
-, automake
-, libtool
+, autoreconfHook
 , pkg-config
 }:
 
@@ -27,14 +25,10 @@ stdenv.mkDerivation rec {
   ];
 
   nativeBuildInputs = [
-    autoconf
-    automake
+    autoreconfHook
     pkg-config
-    libtool
   ];
 
-  preConfigure = "./bootstrap";
-
   meta = with lib; {
     homepage = "https://notabug.org/g0tsu/litemdview";
     description = "A suckless markdown viewer";
```
With this change, I think it should be good to go, unless you want to add any `passthru.tests`. Hard to test a GUI app, though.

File: pkgs/applications/graphics/litemdview/default.nix
Comment (WhiteBlackGoose) [reply pkgs/applications/graphics/litemdview/default.nix]: Done, thanks!

**Comment by WhiteBlackGoose** [comment]
Note: I keep all my package in one commit, is it how it should be, or should I commit changes per PR review in different commits?

**Comment by NickCao** [comment]
There's a conflict in maintainers/maintainer-list.nix, also fetchpatch appears unused.

**Comment by NickCao** [comment]
The best way is to split it into two commits:
- maintainers: add WhiteBlackGoose
-  litemdview: init at 0.0.32

**Comment by WhiteBlackGoose** [comment]
Oh yeah that's what I meant (and how I keep it). Okay thanks
I'll quickly remove the unused dep and rebase

**Comment by NickCao** [comment]
@ofborg build litemdview

**Comment by tejing1** [comment]
Result of `nixpkgs-review pr 209770` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>litemdview</li>
  </ul>
</details>

**Review by tejing1 — APPROVED** [review]

File: pkgs/applications/graphics/litemdview/default.nix
Comment (NickCao) [inline pkgs/applications/graphics/litemdview/default.nix:57]:
```suggestion
    platforms = platforms.linux;
```

Looks like the build is failing on darwin?

Comment (tejing1) [reply pkgs/applications/graphics/litemdview/default.nix:57]: Is it? the darwin tests show as queued for me. They normally take longer to get around to running.

Comment (WhiteBlackGoose) [reply pkgs/applications/graphics/litemdview/default.nix:57]: Oops, thanks! Is it possible for me to check if it builds on darwin before someone else checks it? Or only with a VM?

Comment (NickCao) [reply pkgs/applications/graphics/litemdview/default.nix:57]: The logs show the opposite: https://logs.nix.ci/?attempt_id=ad66df05-df39-409a-b0ed-a01442b7d0e9&key=nixos%2Fnixpkgs.209770

Comment (NickCao) [reply pkgs/applications/graphics/litemdview/default.nix]: > Oops, thanks! Is it possible for me to check if it builds on darwin before someone else checks it? Or only with a VM?

Just leave it to @ofborg, the bot that does exactly the job.

**Review by NickCao — APPROVED** [review]

**Pull request merged**
